intuita-webview/src/codemodList/App.tsx

```diff
@@ -13,55 +12,19 @@ const setSearchPhrase = (searchPhrase: string) => {
 	});
 };
 
-function App() {
-	const [view, setView] = useState({
-		viewProps: window.INITIAL_STATE.codemodListProps,
-		viewId: 'codemods',
-	});
-
-	useEffect(() => {
-		const handler = (e: MessageEvent<WebviewMessage>) => {
-			const message = e.data;
-			if (
-				message.kind === 'webview.codemodList.setView' &&
-				message.value.viewId === 'codemods'
-			) {
-				setView(message.value);
-			}
-		};
-
-		window.addEventListener('message', handler);
-
-		return () => {
-			window.removeEventListener('message', handler);
-		};
-	}, []);
-
-	if (view === null) {
-		return (
-			<main className="App">
-				<Progress />
-			</main>
-		);
-	}
-
-	const { codemodTree, autocompleteItems, searchPhrase, rootPath } =
-		view.viewProps;
-
-	return (
+export const App = memo(
+	(props: MainWebviewViewProps & { activeTabId: 'codemods' }) => (
```

Comment:
if we pass props to the App component `memo` will not work without comparator, because props are nested object